### PR TITLE
Wip2

### PR DESCRIPTION
**Pull request by tobias-**

### CODE REVIEW

File: src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java [hunk 1]
```diff
@@ -160,7 +160,12 @@ private void doProcess() throws IOException, IllegalClassFormatException {
       File file = entry.getValue();
 
       progressIndicator.setText2(className);
+
+      classBytesReader.setSearchScopeFromFile(file);
```
Comment (pgilmore) [inline src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java]: Why do we need to scope the search?

File: src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java [hunk 2]
```diff
@@ -174,7 +179,7 @@ private void processEnhancement(IdeaClassLoader classLoader, Transformer transfo
       byte[] transformed = transformer.transform(classLoader, className, null, null, origBytes);
       if (transformed != null) {
         writeTransformed(file, transformed);
-        logInfo("enhanced: " + className);
+        //logInfo("enhanced: " + className);
```
Comment (pgilmore) [inline src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java]: Log trace ?

File: src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java [hunk 3]
```diff
@@ -212,8 +217,13 @@ private void logError(String msg) {
   private void addFileSystemUrl(List<URL> out, VirtualFile outDir) throws MalformedURLException {
     if (outDir != null) {
       String url = outDir.getUrl();
-      // take into account windows file system
-      url = url.replace("file://", "file:/");
+      if (outDir.isDirectory() && !url.endsWith("/")) {
+        url = url + "/";
+      }
+      if ('\\' == File.separatorChar) {
```
Comment (pgilmore) [inline src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java]: :+1:

File: src/main/java/io/ebean/idea/ebean10/plugin/IdeaClassBytesReader.java [hunk 4]
```diff
@@ -111,59 +128,63 @@ public IdeaClassBytesReader(CompileContext compileContext, Map<String, File> com
...
   }
 
-  private VirtualFile getClassFileFromModuleTestOutput(Module module, String classNamePathWithExtension) {
-    // Search the file in the module's test output directory.
-    final VirtualFile outputDirectory = compileContext.getModuleOutputDirectoryForTests(module);
-    if (outputDirectory == null) {
-      warn(module.getName() + ": Couldn't find test output directory!");
-      return null;
+  private GlobalSearchScope findModule(final VirtualFile virtualFile) {
+    for (Module module : compileContext.getCompileScope().getAffectedModules()) {
```
Comment (pgilmore) [inline src/main/java/io/ebean/idea/ebean10/plugin/IdeaClassBytesReader.java]: You lost me at PSI..

File: src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java (hunk 2 above)
Comment (tobias-) [reply src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java]: Thanks. Need to revert/remove that patch before I do the PR. The plugin is extremely verbose imho, so I cut it down to see problems.

File: src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java (hunk 3 above)
Comment (tobias-) [reply src/main/java/io/ebean/idea/ebean10/plugin/EbeanEnhancementTask.java]: Yeah, I'm puking inside... Not sure if that code actually is good even on windows. Shouldn't be imho

File: src/main/java/io/ebean/idea/ebean10/plugin/IdeaClassBytesReader.java (hunk 4 above)
Comment (tobias-) [reply src/main/java/io/ebean/idea/ebean10/plugin/IdeaClassBytesReader.java]: PSI is kind of like AST but afaik with styling preserved.

Comment (tobias-) [reply src/main/java/io/ebean/idea/ebean10/plugin/IdeaClassBytesReader.java]: Btw, you do not want to know how many hours I've spent trying to find examples or documentation about this.

**Pull request closed without merging**